homeassistant/components/rflink/sensor.py

```diff
@@ -73,9 +142,12 @@ def devices_from_config(domain_config):
     devices = []
     for device_id, config in domain_config[CONF_DEVICES].items():
         if ATTR_UNIT_OF_MEASUREMENT not in config:
-            config[ATTR_UNIT_OF_MEASUREMENT] = lookup_unit_for_sensor_type(
-                config[CONF_SENSOR_TYPE]
-            )
+            if (
+                config[CONF_SENSOR_TYPE] not in SENSOR_TYPES_DICT
+            ):  # only if not SensorEntityDescription
+                config[ATTR_UNIT_OF_MEASUREMENT] = lookup_unit_for_sensor_type(
```

Comment:
While this is old code. It mutates the config, which is somewhat vad.
Cant we lookup the unit inside constructor instead if passed unit is none and there is no entity description. Alternatively construct an entity description on the fly using this unit lookup.

Reply:
Done